### PR TITLE
Adds check to make sure BLUESPAWN runs on matching arch

### LINKED ISSUE #303: Add checks to ensure that BLUESPAWN client is running on match OS arch

### PR DESCRIPTION
**Pull request by ION28**
Closes #303

### CODE REVIEW

**Comment by ION28** [comment]
Tested all combinations (x64 on x64, x64 on x86, x86 on x86, and x86 on x64) and behaves as desired. Note I used https://docs.microsoft.com/en-us/windows/win32/api/wow64apiset/nf-wow64apiset-iswow64process instead of IsWow64Process2 because IsWow64Process2 is only win10/16+

**Comment by Jack-McDowell** [comment]
Make sure to link dynamically!

File: BLUESPAWN-client/BLUESPAWN-client.vcxproj [hunk 1]
```diff
@@ -302,7 +329,55 @@
     </ClCompile>
     <Link>
       <AdditionalDependencies>Secur32.lib;DbgHelp.lib;Wintrust.lib;ws2_32.lib;Crypt32.lib;Shlwapi.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <EnableUAC Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
```
Comment (Jack-McDowell) [inline BLUESPAWN-client/BLUESPAWN-client.vcxproj:332]: Remove the condition statement and make this more concise

File: BLUESPAWN-client/BLUESPAWN-client.vcxproj [hunk 2]
```diff
@@ -302,7 +329,55 @@
...
+      <UACUIAccess Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
+      </UACUIAccess>
+      <EnableUAC Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+      </EnableUAC>
+      <UACExecutionLevel Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+      </UACExecutionLevel>
+      <UACUIAccess Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+      </UACUIAccess>
     </Link>
+    <PostBuildEvent>
```
Comment (Jack-McDowell) [inline BLUESPAWN-client/BLUESPAWN-client.vcxproj:357]: Remove the condition statement and make this more concise

File: BLUESPAWN-client/src/user/BLUESPAWN.cpp [hunk 3]
```diff
@@ -173,12 +174,40 @@ void print_help(cxxopts::ParseResult result, cxxopts::Options options) {
...
+void Bluespawn::check_correct_arch() {
+	BOOL bIsWow64 = FALSE;
+	if (IsWindows10OrGreater()) {
+		typedef BOOL (WINAPI *LPFNIsWow64Process2) (HANDLE, PUSHORT, PUSHORT);
+
+		LPFNIsWow64Process2 fnIsWow64Process2;
+		USHORT ProcessMachine;
+		USHORT NativeMachine;
+
+		fnIsWow64Process2 = (LPFNIsWow64Process2)GetProcAddress(GetModuleHandle(TEXT("kernel32")), "IsWow64Process2");
```
Comment (Jack-McDowell) [inline BLUESPAWN-client/src/user/BLUESPAWN.cpp:186]: reinterpret_cast.

Also, see the linker in commonlib

File: BLUESPAWN-client/BLUESPAWN-client.vcxproj (hunk 1 above)
Comment (ION28) [reply BLUESPAWN-client/BLUESPAWN-client.vcxproj:332]: deleted entirely as the full manifest is specified in the file and not generated

File: BLUESPAWN-client/BLUESPAWN-client.vcxproj (hunk 2 above)
Comment (ION28) [reply BLUESPAWN-client/BLUESPAWN-client.vcxproj:357]: fixed

File: BLUESPAWN-client/src/user/BLUESPAWN.cpp (hunk 3 above)
Comment (ION28) [reply BLUESPAWN-client/src/user/BLUESPAWN.cpp]: fixed by using DynamicLinker I think

**Pull request merged**